src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurCashAdvancePdpFeedFileServiceImpl.java

```diff
@@ -1,36 +1,299 @@
+import edu.cornell.kfs.concur.batch.xmlObjects.PdpFeedDetailEntry;
+import edu.cornell.kfs.concur.batch.xmlObjects.PdpFeedFileBaseEntry;
+import edu.cornell.kfs.concur.batch.xmlObjects.PdpFeedGroupEntry;
+import edu.cornell.kfs.concur.batch.xmlObjects.PdpFeedHeaderEntry;
+import edu.cornell.kfs.concur.batch.xmlObjects.PdpFeedPayeeIdEntry;
+import edu.cornell.kfs.concur.batch.xmlObjects.PdpFeedTrailerEntry;
+import edu.cornell.kfs.sys.CUKFSConstants;
+import edu.cornell.kfs.sys.CUKFSParameterKeyConstants;
+import edu.cornell.kfs.sys.service.CUMarshalService;
 
-public class ConcurCashAdvancePdpFeedFileServiceImpl extends InitiateDirectoryBase implements ConcurCashAdvancePdpFeedFileService {
+public class ConcurCashAdvancePdpFeedFileServiceImpl implements ConcurCashAdvancePdpFeedFileService {
     private static org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(ConcurCashAdvancePdpFeedFileServiceImpl.class);
-    protected String outgoingDirectoryName;
+    private int payeeNameFieldSize = 0;
+    protected String paymentImportDirectory;
+    protected DateTimeService DateTimeService;
+    protected DataDictionaryService dataDictionaryService;
+    protected ConcurRequestedCashAdvanceService concurRequestedCashAdvanceService;
+    protected CUMarshalService cuMarshalService;
+    protected ParameterService parameterService;
+
+    @Transactional
+    public boolean createPdpFeedFileForValidatedDetailFileLines(ConcurRequestExtractFile requestExtractFile) {
+        boolean pdpFileSuccessfullyCreated = false;
+        PdpFeedFileBaseEntry pdpFeedFileDataObject = buildPdpFeedBaseEntry(requestExtractFile);
+        if (pdpFeedFileDataObject.getTrailer().getDetailCount().intValue() != 0) {
+            String fullyQualifiedPdpFileName = buildFullyQualifiedPdpOutputFileName(getPaymentImportDirectory(), requestExtractFile.getFileName());
+            pdpFileSuccessfullyCreated = createPdpFeedFile(pdpFeedFileDataObject, fullyQualifiedPdpFileName);
+            if (pdpFileSuccessfullyCreated) {
+                LOG.debug("fullyQualifiedPdpFileName [" + fullyQualifiedPdpFileName + "]  was created for requestExtractFile [" + requestExtractFile.getFileName() + "]");
+                requestExtractFile.setFullyQualifiedPdpFileName(fullyQualifiedPdpFileName);
+            }
+            else {
+                LOG.debug("FAILED TO CREATE: fullyQualifiedPdpFileName [" + fullyQualifiedPdpFileName + "] for requestExtractFile [" + requestExtractFile.getFileName() + "]");
+            }
+        }
+        return pdpFileSuccessfullyCreated;
+    }
+
+    public void createDoneFileForPdpFile(String fullyQualifiedPdpFileName) throws IOException {
+        String fullFilePath = StringUtils.replace(fullyQualifiedPdpFileName, ConcurConstants.XML_FILE_EXTENSION, GeneralLedgerConstants.BatchFileSystem.DONE_FILE_EXTENSION);
+        FileUtils.touch(new File(fullFilePath));
+    }
+
+    public void setPaymentImportDirectory(String paymentImportDirectory) {
```

Comment:
I will move the getter/setters to the end at your request.  The pattern I was following was to have all public methods first and private/protected methods after them.